pkg/packageGenerator/csharp/generator.go

```diff
@@ -0,0 +1,72 @@
+package csharp
+
+import (
+	"fmt"
+	"github.com/pkg/errors"
+	"path/filepath"
+	"spring-financial-group/jx3-openapi-generation/pkg/commandRunner"
+	"spring-financial-group/jx3-openapi-generation/pkg/domain"
+	"spring-financial-group/jx3-openapi-generation/pkg/file"
+)
+
+const (
+	NugetConfigPath = "./registry/nuget.config"
+)
+
+type Generator struct {
+	Version     string
+	ServiceName string
+	RepoOwner   string
+	RepoName    string
+
+	Cmd    domain.CommandRunner
+	FileIO domain.FileIO
+}
+
+func NewGenerator(version, name, repoOwner, repoName string) domain.PackageGenerator {
+	return &Generator{
+		Version:     version,
+		ServiceName: name,
+		RepoOwner:   repoOwner,
+		RepoName:    repoName,
+		Cmd:         commandRunner.NewCommandRunner(),
+		FileIO:      file.NewFileIO(),
+	}
+}
+
+func (g *Generator) GeneratePackage(specificationPath, outputDir string) (string, error) {
+	packageDir, err := g.FileIO.MkdirAll(filepath.Join(outputDir, g.GetPackageName()), 0755)
+	if err != nil {
+		return "", err
+	}
+
+	// Generate Package
+	err = g.Cmd.ExecuteAndLog(outputDir, "npx", "openapi-generator-cli", "generate",
```

Comment:
Could definitely do this with a base config or something I think, sems like a lot of duplication?